timemachine/cpp/src/kernels/k_barostat.cuh

```diff
@@ -92,23 +92,25 @@ template <typename RealType>
 void __global__ k_setup_barostat_move(
     const bool adaptive,
     const RealType *__restrict__ rand,     // [2], use first value, second value is metropolis condition
-    double *__restrict__ d_box,            // [3*3]
+    const double *__restrict__ d_box,      // [3*3]
     RealType *__restrict__ d_volume_delta, // [1]
     double *__restrict__ d_volume_scale,   // [1]
-    RealType *__restrict__ d_length_scale  // [1]
+    RealType *__restrict__ d_length_scale, // [1]
+    RealType *__restrict__ d_volume        // [1]
 ) {
     const int idx = blockIdx.x * blockDim.x + threadIdx.x;
     if (idx >= 1) {
         return; // Only a single thread needs to perform this operation
     }
     const RealType volume = d_box[0 * 3 + 0] * d_box[1 * 3 + 1] * d_box[2 * 3 + 2];
-    if (d_volume_scale[0] == 0.0 && adaptive) {
-        d_volume_scale[0] = 0.01 * volume;
-    }
     const RealType delta_volume = d_volume_scale[0] * 2 * (rand[0] - 0.5);
     const RealType new_volume = volume + delta_volume;
-    d_volume_delta[0] = delta_volume;
-    d_length_scale[0] = cbrt(new_volume / volume);
+    if (adaptive && *d_volume_scale == 0.0) {
```

Comment:
is this reordering intended?


if:
1) `adaptive == True`
2) `d_volume_scale[0] == 0.0`
3) `volume == 1.0`

before:

`const RealType delta_volume = 0.01  * 2 * (rand[0] - 0.5);` # this is not zero

after:

`const RealType delta_volume = 0  * 2 * (rand[0] - 0.5);` # this is zero

Reply:
off topic, but does it make sense for pre-commit to run a `make clean && make`? (or at least a `make`)

Reply:
Posting offline resolution of pre-commit issue for reference: we currently do not check that the C++ code still compiles with just `pre-commit run` (unless `wrap_kernels.cpp` was modified in the diff). (`pre-commit run --all-files`, which is run by CI, should catch this).